src/node.cc

```diff
@@ -4322,6 +4401,26 @@ static void StartNodeInstance(void* arg) {
       } while (more == true);
     }
 
+    Local<Value> promise_keys_set =
+        env->promise_unhandled_reject_keys().As<Value>();
+    Local<Function> convert = env->array_from();
+    Local<Value> ret = convert->Call(env->context(),
+        Null(env->isolate()), 1, &promise_keys_set).ToLocalChecked();
+    Local<Array> promise_keys = ret.As<Array>();
+    uint32_t key_count = promise_keys->Length();
+    Local<NativeWeakMap> unhandled_reject_map =
+        env->promise_unhandled_reject_map();
+
+    for (uint32_t key_iter = 0; key_iter < key_count; key_iter++) {
+      Local<Value> key = promise_keys->Get(env->context(),
+                                           key_iter).ToLocalChecked();
+
+      if (unhandled_reject_map->Has(key)) {
+        ReportPromiseRejection(isolate, unhandled_reject_map->Get(key));
+        exit(1);
```

Comment:
ditto to @bnoordhuis' comment. it should at minimum follow the same path as unhandled exceptions. e.g.
```js
process.on('exit', () => console.error('bye'));
throw new Error('crap');
// output: bye
```